server/src/main/java/io/envoyproxy/controlplane/server/DiscoveryServer.java

```diff
@@ -146,6 +146,18 @@ public void onNext(DiscoveryRequest request) {
             nonce,
             request.getVersionInfo());
 
+        //TODO(ramachavali): we should implement back-off strategy here.
```

Comment:
Let's limit the api update PRs to *only* api updates and any breaking changes. Anything else should be in a separate PR.